evalml/automl/automl_search.py

```diff
@@ -607,10 +607,18 @@ def _compute_cv_scores(self, pipeline, X, y):
         start = time.time()
         cv_data = []
         logger.info("\tStarting cross validation")
+        warnings.filterwarnings("ignore", lineno=665)
         for i, (train, test) in enumerate(self.data_split.split(X, y)):
             logger.debug(f"\t\tTraining and scoring on fold {i}")
             X_train, X_test = X.iloc[train], X.iloc[test]
             y_train, y_test = y.iloc[train], y.iloc[test]
+            if self.problem_type in [ProblemTypes.BINARY, ProblemTypes.MULTICLASS]:
```

Comment:
What do you mean by `TrainingValidationSplit` isn't meant to pass this check? Isn't that the case for `StratefiedKFold` too. I think it's good if this fails for `TrainingValidationSplit`.

Reply:
@bchen1116 and I discussed offline and we think it's ok if this error is raised with a `TrainingValidationSplit` since it means either the training or validation set has only one class.